### DISCUSSION TITLE
RFE: Troublesome begging for mod story extension for serpent isle

### DISCUSSION THREAD

**Discussion by i30817** [exult/exult#471] [General]
;)

Anyway, i always found it strange that serpent isle and blackgate have extra spellbooks. I'd basically like it if the 'spellcasting' AI that was done for the keyring mod could apply to other characters that have a spellbook and magic capability. I'd also find it interesting if you could use transcribe _on_ a spellbook (from another spellbook), to copy (not sure if destructively, tending to not) from a book to another, to 'fill' missing spells without having to buy them again.

Of course, in blackgate/serpent isle i think that there are few characters with magic capability (keyring new party npc excepted, which i think comes with her own spellbook anyway), and usually those you'd expect in serpent isle like Mortegro, Petra and Stefano are strictly temporary, most unfortunately, so i'm not sure it's a cool idea if no npc exists for it to be used on for the whole game, or even more than one dungeon.

**Comment by i30817** [comment]
Mmm seems i was mistaken about serpent isle. There, the only game with the transcribe spell, there is only one spellbook, although there is a ruined one that explodes and another that is lost during the storm but never recovered (although i suppose it could be added in a mod somewhere).

**Comment by KnightCaptainU7** [comment]
Until I looked into this (thanks to your post) I had assumed the spellbook was a "universal item" as was the list of spells in it. Yet adding new spellbooks to the game world makes new blank ones. So adding a spellbook for other party members could be possible.

Transcribe is limited by this line:
UI_find_object(PARTY, SHAPE_SPELLBOOK, ANY, ANY);
So a mod to change that could allow for different spellbooks.

The plans for my mod including adding Andrio to the party, and him having a spellbook would make the sense.
![image](https://user-images.githubusercontent.com/117867844/234452426-7f6d1aed-5b2c-438d-b769-46a617770e7f.png)

My thought was to give him "Andrio's Wand" which when used as a weapon would trigger EVENT == WEAPON. That would use a die_roll based on his level, and then select a spell shape based on that. It would not take into account his Mana level (or just keep working even if the level is 0). I'm not sure how to implement the attack, since spell shapes cannot attack inside a container.

Maybe the act of swinging the wand would swap out the wand in-hand for a moment, put in the intended shape, and then swap it back immediately. That could lead to pausing (or using Vibrate) at the correct moment allowing for the spell to be stolen? Or would the spell disappear because it decreases by Quality until it disappears?

Results by die_roll vs. (level -1) would be:
2nd Cold Blast (100% chance) (I need to add something to SIfixes for this, it shouldn't be a red flame. And I have the artwork done.)
3nd Curse and Paralyze (50% chance this circle, then 50% either spell), and above spells.
4th Flash (33%) and above.
5th, Conjure (25%) and above. Explosion would be risky, could cause friendly fire.
6th, Cold Strike (20%) and above.
7th, Lightning (16.7%)  and above.
8th, Swordstrike (14.7%) and above.
9th, Spiral Missile (12.5%) and above.

Not sure if UI_fire_projectile would help. It doesn't seem to be used in SI/SS.

I have not played through the Keyring Mod myself, but it appears to use a conversation-based system:
https://github.com/exult/exult/blob/master/content/bgkeyring/src/items/spell_items.uc
Playing that should spark some ideas.

**Comment by marzojr** [comment]
UI_fire_projectile is not *used* in SI, but the intrinsic exists there (even in the original game). And I would welcome any graphical fixes to SI Fixes 🙂

**Comment by KnightCaptainU7** [comment]
I need to dig through some of the Avatar Pack shapes, since it looks like the SI shapes 1024 to 1035 are set pairs for the Avatar (with cape) and then shapenumber+6 to get the naked version. To use that same logic for the redheaded versions from the Avatar Pack, I'd want to put them in as specific numbers in that 1000 block.

Would you be okay with me submitting things starting at shape 1100? I want to keep my mod as in-sync as possible with SIfixes without going into comparison hell.

**Comment by marzojr** [comment]
Note that the association of avatar clothed shape/naked shape is not hard-coded in Exult; you must specify it in `avatar_data.txt` file, including also the paperdoll data to use (see [Keyring's version](https://github.com/exult/exult/blob/master/content/bgkeyring/data/avatar_data.txt)).

I really need to make some dialogs in ES to edit all those config files...

**Comment by KnightCaptainU7** [comment]
Okay, I'll have to check the Usecode as well. I'm hoping they didn't
hardcode there, say for Frigidazzi's cutscene.
